### PR TITLE
Ignore symLinks for seacrhing qmake

### LINKED ISSUE #731: Warning: Run qmake file! execve: Permission denied
**Qt libraries no more copied to the assembly.
Showing the warning "Warning: Run qmake file! execve: Permission denied"**

_CQtDeployer version  : 1.6.2227.e8789df
Linux Lenovo-ipg3 5.15.0-52-generic #58-Ubuntu SMP Thu Oct 13 08:03:55 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux_

> cqtdeployer -bin examed deploySystem noTranslations
> "Info: Deploy ..."
> "Info: The targetDir option is not used. CQtDeployer will use default target dir :/media/avttrue/S860/Projects/Projects_QT/Viola/build-examedit-Desktop_Qt_6_2_0_GCC_64bit-Release/DistributionKit"
> "Info: The qmake was found in the PATH variable. qmake : /var/lib/snapd/hostfs/usr/bin/qmake If you want to disable search qmake executable in PATH variable, use the noCheckPATH option"
> "Warning: Run qmake fail! execve: Permission denied"
> "Warning: Failed to execute the qmake process! Trying to initialize Qt directories from path: /var/lib/snapd/hostfs/usr"
> "Info: copy :/media/avttrue/S860/Projects/Projects_QT/Viola/build-examedit-Desktop_Qt_6_2_0_GCC_64bit-Release/examed"
> "Info: copy :/var/lib/snapd/hostfs/lib/x86_64-linux-gnu/libXau.so.6"
> "Info: copy :/var/lib/snapd/hostfs/lib/x86_64-linux-gnu/libbsd.so.0"
> "Info: copy :/var/lib/snapd/hostfs/lib/x86_64-linux-gnu/libxcb.so.1"
> "Info: copy :/var/lib/snapd/hostfs/lib/x86_64-linux-gnu/libGLdispatch.so.0"
> "Info: copy :/var/lib/snapd/hostfs/lib/x86_64-linux-gnu/libXdmcp.so.6"
> "Info: copy :/var/lib/snapd/hostfs/lib/x86_64-linux-gnu/libgcc_s.so.1"
> "Info: copy :/var/lib/snapd/hostfs/lib/x86_64-linux-gnu/libX11.so.6"
> "Info: copy :/var/lib/snapd/hostfs/lib/x86_64-linux-gnu/libOpenGL.so.0"
> "Info: copy :/var/lib/snapd/hostfs/lib/x86_64-linux-gnu/libstdc++.so.6"
> "Info: copy :/var/lib/snapd/hostfs/lib/x86_64-linux-gnu/libGLX.so.0"
> "Info: deploy done!"
> "Info: copy :/media/avttrue/S860/Projects/Projects_QT/Viola/build-examedit-Desktop_Qt_6_2_0_GCC_64bit-Release/DistributionKit/tmp_data/Application/bin/examed"
> "Info: copy :/media/avttrue/S860/Projects/Projects_QT/Viola/build-examedit-Desktop_Qt_6_2_0_GCC_64bit-Release/DistributionKit/tmp_data/Application/bin/qt.conf"
> "Info: copy :/media/avttrue/S860/Projects/Projects_QT/Viola/build-examedit-Desktop_Qt_6_2_0_GCC_64bit-Release/DistributionKit/tmp_data/Application/examed.sh"
> "Info: copy :/media/avttrue/S860/Projects/Projects_QT/Viola/build-examedit-Desktop_Qt_6_2_0_GCC_64bit-Release/DistributionKit/tmp_data/Application/lib/systemLibs/libbsd.so.0"
> "Info: copy :/media/avttrue/S860/Projects/Projects_QT/Viola/build-examedit-Desktop_Qt_6_2_0_GCC_64bit-Release/DistributionKit/tmp_data/Application/lib/systemLibs/libgcc_s.so.1"
> "Info: copy :/media/avttrue/S860/Projects/Projects_QT/Viola/build-examedit-Desktop_Qt_6_2_0_GCC_64bit-Release/DistributionKit/tmp_data/Application/lib/systemLibs/libGLdispatch.so.0"
> "Info: copy :/media/avttrue/S860/Projects/Projects_QT/Viola/build-examedit-Desktop_Qt_6_2_0_GCC_64bit-Release/DistributionKit/tmp_data/Application/lib/systemLibs/libGLX.so.0"
> "Info: copy :/media/avttrue/S860/Projects/Projects_QT/Viola/build-examedit-Desktop_Qt_6_2_0_GCC_64bit-Release/DistributionKit/tmp_data/Application/lib/systemLibs/libOpenGL.so.0"
> "Info: copy :/media/avttrue/S860/Projects/Projects_QT/Viola/build-examedit-Desktop_Qt_6_2_0_GCC_64bit-Release/DistributionKit/tmp_data/Application/lib/systemLibs/libstdc++.so.6"
> "Info: copy :/media/avttrue/S860/Projects/Projects_QT/Viola/build-examedit-Desktop_Qt_6_2_0_GCC_64bit-Release/DistributionKit/tmp_data/Application/lib/systemLibs/libX11.so.6"
> "Info: copy :/media/avttrue/S860/Projects/Projects_QT/Viola/build-examedit-Desktop_Qt_6_2_0_GCC_64bit-Release/DistributionKit/tmp_data/Application/lib/systemLibs/libXau.so.6"
> "Info: copy :/media/avttrue/S860/Projects/Projects_QT/Viola/build-examedit-Desktop_Qt_6_2_0_GCC_64bit-Release/DistributionKit/tmp_data/Application/lib/systemLibs/libxcb.so.1"
> "Info: copy :/media/avttrue/S860/Projects/Projects_QT/Viola/build-examedit-Desktop_Qt_6_2_0_GCC_64bit-Release/DistributionKit/tmp_data/Application/lib/systemLibs/libXdmcp.so.6"

### PR DESCRIPTION
**Pull request by EndrII**
fix #731

### CODE REVIEW

File: src/Deploy/src/configparser.cpp [hunk 1]
```diff
@@ -1135,25 +1135,51 @@ bool ConfigParser::initQmakePrivate(const QString &qmake) {
     QFileInfo info(qmake);
 
     QString basePath = info.absolutePath();
-    if (!setQmake(qmake)) {
-        QDir dir(basePath);
 
-        if (!dir.cdUp()) {
-            QuasarAppUtils::Params::log("Failed to initialize qt directories by qmake.",
+    // Invoke qmake executable only when qmake pathe exclude snapRootFS path.
```
Comment (EndrII) [inline src/Deploy/src/configparser.cpp:1139]:
```suggestion
    // Invoke qmake executable only when qmake paths exclude snapRootFS path.
```

File: src/Deploy/src/configparser.cpp [hunk 2]
```diff
@@ -1135,25 +1135,51 @@ bool ConfigParser::initQmakePrivate(const QString &qmake) {
...
+
+        QString debianQtRoot = QString("/usr/lib/%0/qt%1").
+                               arg(neededPlatform).arg(qtVersion);
+
+
+        if (!setQtDir(debianQtRoot)) {
+            QuasarAppUtils::Params::log("Failed to initialize Qt directories",
                                         QuasarAppUtils::Error);
             return false;
         }
```
Comment (EndrII) [inline src/Deploy/src/configparser.cpp:1161]:
```suggestion
        }
        
        return true;
```

File: src/Deploy/src/configparser.cpp [hunk 3]
```diff
@@ -1135,25 +1135,51 @@ bool ConfigParser::initQmakePrivate(const QString &qmake) {
...
+    if (!qmake.contains(DeployCore::snapRootFS()) && setQmake(qmake)) {
+        return true;
+    }
+
+
+    // try deploy qt using only read FS permisions
+
+    // check debian qt structure
+    if (DeployCore::isDebianQt(info.absoluteFilePath())) {
+        QString neededPlatform = DeployCore::getPlatformLibPrefix(_config.getPlatformOfAll());
```
Comment (EndrII) [inline src/Deploy/src/configparser.cpp:1150]:
```suggestion
        // initialize qt that was installed from apt package manager on any debian based os
        QString neededPlatform = DeployCore::getPlatformLibPrefix(_config.getPlatformOfAll());
```

File: src/Deploy/src/configparser.cpp [hunk 4]
```diff
@@ -1135,25 +1135,51 @@ bool ConfigParser::initQmakePrivate(const QString &qmake) {
...
+                               arg(neededPlatform).arg(qtVersion);
+
+
+        if (!setQtDir(debianQtRoot)) {
+            QuasarAppUtils::Params::log("Failed to initialize Qt directories",
                                         QuasarAppUtils::Error);
             return false;
         }
+    }
+
```
Comment (EndrII) [inline src/Deploy/src/configparser.cpp:1163]:
```suggestion
    // initialize qt that was installed from qt installer or aqtinstall package manager
```

File: src/Deploy/src/configparser.cpp [hunk 5]
```diff
@@ -1135,25 +1135,51 @@ bool ConfigParser::initQmakePrivate(const QString &qmake) {
...
+    QDir dir(basePath);
 
+    if (!dir.cdUp()) {
+        QuasarAppUtils::Params::log("Failed to initialize qt directories by qmake.",
+                                    QuasarAppUtils::Error);
+        return false;
+    }
+
+    // For snap package of cqtdeplyer it is norma behavior
```
Comment (EndrII) [inline src/Deploy/src/configparser.cpp:1172]:
```suggestion
    // For snap package of cqtdeplyer it is normal behavior
```

**Comment by EndrII** [comment]
@avttrue please review this pull request.

**Review by avttrue — APPROVED** [review]

**Review by IgorekLoschinin — APPROVED** [review]

**Pull request merged**